src/Symfony/Component/ObjectMapper/ObjectMapper.php

```diff
@@ -167,7 +168,15 @@ public function map(object $source, object|string|null $target = null): object
 
     private function getRawValue(object $source, string $propertyName): mixed
     {
-        return $this->propertyAccessor ? $this->propertyAccessor->getValue($source, $propertyName) : $source->{$propertyName};
+        if ($this->propertyAccessor) {
+            return $this->propertyAccessor->getValue($source, $propertyName);
+        }
+
+        if (!property_exists($source, $propertyName) && !(method_exists($source, '__isset') && true === $source->__isset($propertyName))) {
```

Comment:
I'm not a huge fan of testing for the `__isset` magic method but if we don't add this its hard to support magic getters properly, maybe we should not support them in the first place? 

Comments welcome.

Reply:
I'm fine not supporting them personally, especially since magic methods don't allow checking for the existence of a property that holds a null value

```suggestion
        if (!property_exists($source, $propertyName) && !isset($source->{$propertyName}))) {
```